evolver/tests/test_device.py

```diff
@@ -126,3 +132,10 @@ def test_non_descriptor_config_field(self):
         assert isinstance(obj.hardware["a"], NoOpSensorDriver)
         assert isinstance(obj.serial, DEFAULT_SERIAL)
         assert isinstance(obj.history, DEFAULT_HISTORY)
+
+    def test_calibration_status(self, demo_evolver):
+        status = demo_evolver.calibration_status
+        assert status.keys() and (status.keys() == demo_evolver.hardware.keys())
+        assert status["testeffector"] is None
+        assert isinstance(status["testsensor"]["input_transformer"], Status)
+        assert isinstance(status["testsensor"]["output_transformer"], Status)
```

Comment:
Looking at this, wondering if it might make sense to have the calibrator have an overall status, so maybe it returns an object (effectively) like:

```py
class CalibratorStatus:
    input_transformer_status: Status
    output_transformer_status: Status
    ok: bool = input.ok and output.ok
```

thoughts?

Reply:
I don't think this can be utilized as a return type for ``app.main.calibration_status``, however, as it can be customized by derived calibrator classes. Since, for example, ``input_transformer`` belongs to the ``Config`` it can be deleted and other fields added, in which case ``Status`` will also need to be updated to account for these new fields. I had originally contemplated this, however the above dissuaded me. This is also why ``Status`` needs to belong to ``Calibration`` rather than at the module level.